wpi.sh

```diff
@@ -0,0 +1,177 @@
+# See the documentation of wpi-many.sh for information on the inputs to this
+# script.
+#
+# Input differences compared to wpi-many.sh:
+# -i and -o are not valid options
+# new required option -d: the absolute path to the directory containing the target project
+#
+
+while getopts "d:u:t:" opt; do
+  case $opt in
+    d) DIR="$OPTARG"
+       ;;
+    u) USER="$OPTARG"
+       ;;
+    t) TIMEOUT="$OPTARG"
+       ;;        
+    \?) # echo "Invalid option -$OPTARG" >&2
+       ;;
+  esac
+done
+
+# shift so that the other arguments (that should be passed to dljc) are all
+# that's in $@
+shift $(( OPTIND - 1 ))
+
+# check required arguments and environment variables:
+
+# Testing for JAVA8_HOME, not a misspelling of JAVA_HOME.
+# shellcheck disable=SC2153
+if [ "x${JAVA8_HOME}" = "x" ]; then
+    echo "JAVA8_HOME must be set to a Java 8 JDK"
+    exit 1
+fi
+
+if [ ! -d "${JAVA8_HOME}" ]; then
+    echo "JAVA8_HOME is set to a non-existent directory ${JAVA8_HOME}"
+    exit 1
+fi
+
+# Testing for JAVA11_HOME, not a misspelling of JAVA_HOME.
+# shellcheck disable=SC2153
+if [ "x${JAVA11_HOME}" = "x" ]; then
+    echo "JAVA11_HOME must be set to a Java 11 JDK"
+    exit 1
+fi
+
+if [ ! -d "${JAVA11_HOME}" ]; then
+    echo "JAVA11_HOME is set to a non-existent directory ${JAVA11_HOME}"
+    exit 1
+fi
+
+JAVA_HOME="${JAVA11_HOME}"
+
+if [ "x${CHECKERFRAMEWORK}" = "x" ]; then
+    echo "CHECKERFRAMEWORK is not set; it must be set to a locally-built Checker Framework. Please clone and build github.com/typetools/checker-framework"
+    exit 2
+fi
+
+if [ ! -d "${CHECKERFRAMEWORK}" ]; then
+    echo "CHECKERFRAMEWORK is set to a non-existent directory ${CHECKERFRAMEWORK}"
+    exit 2
+fi
+
+if [ "x${DIR}" = "x" ]; then
+    echo "wpi.sh was called without a -d argument. The -d argument must be the absolute path to the directory containing the project on which to run WPI."
+    exit 4
+fi
+
+if [ ! -d "${DIR}" ]; then
+    echo "wpi.sh's -d argument was not a directory: ${DIR}"
+    exit 4
+fi
+
+function configure_and_exec_dljc {
+
+  if [ -f build.gradle ]; then
+      if [ -f gradlew ]; then
+	  chmod +x gradlew
+	  GRADLE_EXEC="./gradlew"
+      else
+	  GRADLE_EXEC="gradle"
+      fi
+      CLEAN_CMD="${GRADLE_EXEC} clean -g .gradle -Dorg.gradle.java.home=${JAVA_HOME}"
+      BUILD_CMD="${GRADLE_EXEC} clean compileJava -g .gradle -Dorg.gradle.java.home=${JAVA_HOME}"
+  elif [ -f pom.xml ]; then
+      if [ -f mvnw ]; then
+	  chmod +x mvnw
+	  MVN_EXEC="./mvnw"
+      else
+	  MVN_EXEC="mvn"
+      fi
+      # if running on java 8, need /jre at the end of this Maven command
+      if [ "${JAVA_HOME}" = "${JAVA8_HOME}" ]; then
+          CLEAN_CMD="${MVN_EXEC} clean -Djava.home=${JAVA_HOME}/jre"
+          BUILD_CMD="${MVN_EXEC} clean compile -Djava.home=${JAVA_HOME}/jre"
+      else
+          CLEAN_CMD="${MVN_EXEC} clean -Djava.home=${JAVA_HOME}"
+          BUILD_CMD="${MVN_EXEC} clean compile -Djava.home=${JAVA_HOME}"
+      fi
+  else
```

Comment:
https://github.com/kelloggm/do-like-javac supports Ant.  Why not include that here?

Reply:
Great question. I probably forgot that do-like-javac supports Ant when writing these scripts.

I'd rather merge this PR as-is rather than adding and testing another kind of functionality. I'll open an issue to support Ant builds, and hopefully be able to address it in a future PR.

Reply:
#15